src/silverlabnwb/metadata_gui.py

```diff
@@ -394,40 +396,55 @@ def record_expt(self):
                 elif field in expt:
                     del expt[field]
 
-    def make_label(self, parent, name, row=0, sticky='w'):
+    def make_label(self, parent, name, row=0, sticky='w', tooltip_text=''):
         """Make the human-friendly label for a form section."""
         name = name.capitalize().replace('_', ' ') + ':'
-        ttk.Label(parent, text=name).grid(row=row, column=0, sticky=sticky)
+        label = ttk.Label(parent, text=name)
+        label.grid(row=row, column=0, sticky=sticky)
+        if len(tooltip_text) > 0:
+            Hovertip(label, tooltip_text, hover_delay=1000)  # delay in ms
 
     def make_expts_part(self, parent, part_name):
         """Make a component frame for the experiment editor."""
         frame = ttk.Frame(parent)
-        self.make_label(frame, part_name)
         if isinstance(self.template_expt[part_name], collections.Mapping):
             # This is actually a related group of fields
+            self.make_label(frame, part_name)
             self.expts_boxes[part_name] = boxes = {}
-            self.make_expts_fields(frame, self.template_expt[part_name], boxes)
+            self.make_expts_fields(frame, self.template_expt[part_name], boxes, part_name)
         else:
+            self.make_label(frame, part_name, tooltip_text=self.template_expt_comments[part_name])
             textbox = T.Text(frame, width=100, height=5, wrap='word')
             textbox.grid(row=1, column=0, sticky='nesw')
-            textbox.insert('1.0', self.template_expt[part_name])
+            expts_part = self.template_expt[part_name]
+            if expts_part is not None:
+                textbox.insert('1.0', expts_part)
+            else:
+                textbox.insert('1.0', "")
             self.expts_boxes[part_name] = textbox
         frame.pack(side='top', fill='x', expand=True)
 
-    def make_expts_fields(self, frame, template, boxes):
+    def make_expts_fields(self, frame, template, boxes, part_name):
         """Create a related group of fields for an experiment section."""
         for i, field in enumerate(template):
-            self.make_label(frame, field, row=i + 1, sticky='ne')
             if isinstance(template[field], collections.Mapping):
+                self.make_label(frame, field, row=i + 1, sticky='ne')
                 # Another nesting level!
                 boxes[field] = {}
                 subframe = ttk.Frame(frame)
                 for j, subfield in enumerate(template[field]):
-                    self.make_label(subframe, subfield, row=j, sticky='e')
+                    self.make_label(subframe, subfield, row=j, sticky='e',
+                                    tooltip_text=self.template_expt_comments[part_name][field][subfield])
```

Comment:
Maybe define `comments = self.template_expt_comments[part_name]` just before this, to look it up only once and shorten the access?
Or even earlier on (maybe line 430), since it's also used in the `else` branch.